File: frontend/src/lib/sheets/mappers/accountBank.mapper.ts
import type { AccountBank } from '../../../types/api';

export type AccountBankSheetRow = Record<string, string | number | boolean>;

export const mapRowToAccountBank = (row: AccountBankSheetRow): AccountBank => ({
  _id: String(row._id ?? ''),
  customerCode: String(row.customerCode ?? ''),
  customerName: String(row.customerName ?? ''),
  bank: String(row.bank ?? ''),
  accountNumber: String(row.accountNumber ?? ''),
  isDefault: row.isDefault === true || row.isDefault === 'TRUE' || row.isDefault === 'true',
});

export const mapAccountBankToRow = (
  bank: Partial<AccountBank> & { createdAt?: string; updatedAt?: string },
): Record<string, unknown> => ({
  _id: bank._id,
  customerCode: bank.customerCode,
  customerName: bank.customerName,
  bank: bank.bank,
  accountNumber: bank.accountNumber,
  isDefault: bank.isDefault ?? false,
  createdAt: bank.createdAt,
  updatedAt: bank.updatedAt,
});

File: mobile/lib/sheets/mappers/account_bank_mapper.dart
import '../../models/api_models.dart';

AccountBank mapRowToAccountBank(Map<String, Object?> row) {
  final isDefaultRaw = row['isDefault'];
  final isDefault = isDefaultRaw == true ||
      isDefaultRaw == 'TRUE' ||
      isDefaultRaw == 'true' ||
      isDefaultRaw == 1;

  return AccountBank(
    id: row['_id']?.toString() ?? '',
    customerCode: row['customerCode']?.toString() ?? '',
    customerName: row['customerName']?.toString() ?? '',
    bank: row['bank']?.toString() ?? '',
    accountNumber: row['accountNumber']?.toString() ?? '',
    isDefault: isDefault,
    createdAt: row['createdAt']?.toString(),
    updatedAt: row['updatedAt']?.toString(),
  );
}

Map<String, Object?> mapAccountBankToRow(AccountBank bank) => {
      '_id': bank.id,
      'customerCode': bank.customerCode,
      'customerName': bank.customerName,
      'bank': bank.bank,
      'accountNumber': bank.accountNumber,
      'isDefault': bank.isDefault,
      'createdAt': bank.createdAt,
      'updatedAt': bank.updatedAt,
    };
